ui/tests/integration/components/kv/page/kv-page-version-history-test.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { setupEngine } from 'ember-engines/test-support';
+import { setupMirage } from 'ember-cli-mirage/test-support';
+import { click, render } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+import { kvMetadataPath } from 'vault/utils/kv-path';
+import { PAGE } from 'vault/tests/helpers/kv/kv-page-selectors';
+import { allowAllCapabilitiesStub } from 'vault/tests/helpers/stubs';
+
+module('Integration | Component | kv | Page::Secret::Version-History', function (hooks) {
+  setupRenderingTest(hooks);
+  setupEngine(hooks, 'kv');
+  setupMirage(hooks);
+
+  hooks.beforeEach(async function () {
+    const store = this.owner.lookup('service:store');
+    this.server.post('/sys/capabilities-self', allowAllCapabilitiesStub());
+    const metadata = this.server.create('kv-metadatum');
+    metadata.id = kvMetadataPath('kv-engine', 'my-secret');
+    store.pushPayload('kv/metadatum', {
+      modelName: 'kv/metadata',
+      ...metadata,
+    });
+    this.metadata = store.peekRecord('kv/metadata', metadata.id);
+    this.breadcrumbs = [
+      { label: 'secrets', route: 'secrets', linkExternal: true },
+      { label: this.metadata.backend, route: 'list' },
+      { label: this.metadata.path, route: 'secret.details', model: this.metadata.path },
+      { label: 'version history' },
+    ];
+  });
+
+  test('it renders version history and shows icons for deleted, destroyed and current', async function (assert) {
+    assert.expect(8);
+
+    await render(
+      hbs`
+       <Page::Secret::VersionHistory
+        @path={{this.metadata.path}}
+        @metadata={{this.metadata}}
+        @breadcrumbs={{this.breadcrumbs}}
+      />
+      `,
+      { owner: this.engine }
+    );
+
+    for (const version in this.metadata.versions) {
+      const data = this.metadata.versions[version];
+      assert.dom(PAGE.list.linkedBlock(version)).exists(`renders the linked blocks for each version`);
+
+      if (data.destroyed) {
+        assert
+          .dom(`${PAGE.list.icon(version)} [data-test-icon="x-square-fill"]`)
+          .hasStyle({ color: 'rgb(199, 52, 69)' });
```

Comment:
Unlike the version drop down view, I have a span around my icons that contain the styling classes. Thus, I couldn't use `hasClass` because the styling class was an element higher than the icon. I need the wrapping span because I have to add style to the word "destroyed" and the icon. So instead I confirm that the icon has the color applied to it.
```
<span class="has-text-danger is-size-8 is-block">
                  <Icon @name="x-square-fill" />Destroyed
                </span>
```